### DISCUSSION TITLE
grocy.update_generic Entity_type not callable

### DISCUSSION THREAD

**Discussion by horphi0815** [SebRut/pygrocy#160] [Support] [answered]
Hi,

i dont get the entity_type called.

The request does not grab a entity.

How can i select the entity_type?

`grocy.update_generic(entity_type="locations" , object_id=7, updated_data={"name": "Le new battery"})`

Many thanks . best regards,

Phil

**Comment by horphi0815** [comment]
Can you assist me in this case? Syntax is unclear for me to trigger the API correctly.

`Traceback (most recent call last):
  File "/PycharmProjects/pythonProject/grocy.py", line 11, in <module>

    grocy.update_generic(entity_type='locations', object_id=9, updated_data={ "id": "8",  "name": "Cookies4"})

  File "r\PycharmProjects\pythonProject\venv\lib\site-packages\pygrocy\grocy.py", line 219, in update_generic
    return self._api_client.update_generic(entity_type.value, object_id, updated_data)

AttributeError: 'str' object has no attribute 'value'`

Best regards,

Phil

**Comment by SebRut** [comment, marked as answer]
Hi,

`update_generic` expects an `EntityType` (as stated in the [docs](https://sebrut.github.io/pygrocy/grocy.html#pygrocy.grocy.Grocy.update_generic) ;)). So you would need to change your code to something like this:

```python
grocy.update_generic(entity_type=EntityType.LOCATIONS, object_id=9, updated_data={ "id": "8",  "name": "Cookies4"})
```

**Reply by SebRut** [reply]
@horphi0815 Is this issue solved?

**Comment by horphi0815** [comment]
Yes it is. Thank you very much. :-)
